### ISSUE TITLE
Importing SVG's in 4.3.dev with commit 73589f6 is crashing

### ISSUE THREAD

**Issue by RevoluPowered** [godotengine/godot#87788]
### Tested versions

- Reverting https://github.com/godotengine/godot/pull/87612 resolves the crashing
- Crash is not always present on windows but sometimes is
- Linux could repro the crash each time booting the engine provided you nuke the .godot/ folder

### System information

Linux - Fedora 38 - As up to date as possible

### Issue description

When importing SVG files in multiple threads the engine crashes with a realloc() crash, I did try to workaround this but had the same crash
![image](https://github.com/godotengine/godot/assets/748770/07c0f438-b138-467a-910b-c081aaf3d491)

### Steps to reproduce

- Import a pool of SVG's (doesn't seem to matter the type)

### Minimal reproduction project (MRP)

[TESTPROJECTREPRO.zip](https://github.com/godotengine/godot/files/14113916/TESTPROJECTREPRO.zip)

Images taken from another source, our project files are completely unrelated and they also crash.

**Comment by akien-mga** [comment]
CC @capnm

**Comment by capnm** [comment]
Unfortunately I can't reproduce this myself, I even tested the PR  importing a large amount of SVGs  (debian 12, Vulkan API 1.3.230 - Forward+):
[svg.zip](https://github.com/godotengine/godot/files/14117328/svg.zip)

I have now run it again on the current tip 9adb7c7 and didn't see any issue 
(with or without
```
[rendering]
driver/threads/thread_model=2
```
)

If anyone is able to reproduce this, perhaps could try to bisect the mentioned full ThorVG changelog ...

**Comment by hermet** [comment]
@RevoluPowered Hello, I'm wondering if you can use the thread sanitizer to catch any clues?

**Comment by Rubonnek** [comment]
If  the project is configured to use single threads (i.e. modify `project.godot` to include):

```ini
[editor]

import/use_multiple_threads=false
```

The SVGs import fine. 

I'm able to reproduce the crash but I don't think it's related to ThorVG -- the crash happens at different places and due to double frees or some memory corruption. The root cause is most likely a race condition.

I stumbled upon a similar issue in #84364 with the same workaround, but it's far easier to reproduce the crash by importing these SVGs.

**Comment by akien-mga** [comment]
@RevoluPowered @Rubonnek I couldn't reproduce this bug on Linux (Mageia 9), but I suspect it's related to #87962 (which I can reproduce) and should be fixed by #88053. Could you test to confirm?

**Comment by Rubonnek** [comment]
@akien-mga can confirm --  I can't reproduce the crash described in this issue anymore.
